### PR TITLE
Reformat code with `cargo fmt`

### PR DESCRIPTION
**Pull request by Nopey**

### CODE REVIEW

**Review by mrobinson — CHANGES_REQUESTED** [review]
Great. I noticed that this needed to be done! Would it be possible to make two changes here:

1. Specify a stable rust version in the `rust-toolchain.toml` and include the `rustfmt` component.
2. Add a CI job to run `rustfmt --check` to make sure this doesn't regress.

Thanks!

**Comment by Nopey** [comment]
Added the CI step & pinned rustfmt version.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -82,3 +82,11 @@ jobs:
       - name: Mark the job as unsuccessful
         if: contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled')
         run: exit 1
+  fmt:
+    name: Rustfmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (mrobinson) [inline .github/workflows/main.yml]: `checkout@v2` uses a deprecated version of node.

```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -82,3 +82,11 @@ jobs:
...
+      - uses: mkroening/rust-toolchain-toml@main
+      - name: Rustfmt
```
Comment (mrobinson) [inline .github/workflows/main.yml]: I think the name is redundant here and you could probably remove it.

**Review by mrobinson — CHANGES_REQUESTED** [review]
Please move the job definition above `build_result` and make sure that `build_result` depends on it. Otherwise, the CI will not fail if `rustfmt --check` fails. 

Otherwise, I have a few nits:

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -82,3 +82,11 @@ jobs:
       - name: Mark the job as unsuccessful
         if: contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled')
         run: exit 1
+  fmt:
+    name: Rustfmt
```
Comment (mrobinson) [inline .github/workflows/main.yml]: This is a nit but `rustfmt` is the name of the tool.

```suggestion
    name: Run `rustfmt`
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -82,3 +82,11 @@ jobs:
       - name: Mark the job as unsuccessful
         if: contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled')
         run: exit 1
+  fmt:
```
Comment (mrobinson) [inline .github/workflows/main.yml]: No need to abbreviate here:

```suggestion
  format:
```

**Review by mrobinson — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 4 above)
Comment (Nopey) [reply .github/workflows/main.yml]: Removed abbreviation, now using `Format:` (with a capital F) to match the Build step's capitalization

**Review by mrobinson — APPROVED** [review]

**Pull request merged**